### PR TITLE
feat: Initial support for SwiftUI

### PR DESCRIPTION
**Pull request by campionfellin**
Hey there! I'd like to start using the BMAC badge in my SwiftUI project, but did not see support for it.

I somewhat implemented it in this PR and it seems to work (in the simulator) with IAP.

Would love some feedback as this would be my first PR on this `Pod` and I'm not sure they style you are looking for. This doesn't implement the `fallback()` mechanism fully and I know there are likely some other issues. I'm not super knowledgeable about Swift idioms but just looking to learn!

Here is a screenshot:

Screen Shot 2020-11-29 at 12 40 53 PM

Let me know if we can collaborate and get this pushed through!

Related to https://github.com/appcraftstudio/buymeacoffee/issues/5

Signed-off-by: campionfellin <campionfellin@gmail.com>

### CODE REVIEW

**Comment by frboulais** [comment]
Hi there :)
Thank you very much for your work, I have to take a look at it because I don't understand why there is so much duplicated code. Why creating new color & font enum for SwiftUI and not extending the existing ones? 
Moreover, regarding your screenshot, I'm not sure the button fit the Buy Me a Coffee guidelines (no round corners, no horizontal insets, and the shadow should be applied on the button itself, not its content.

**Comment by campionfellin** [comment]
Hey! Thanks for taking a look!

I haven't spent a ton of time developing with Swift, so I fully expect there to be issues.

I'll start with your suggestion about fixing the Color and Font enums so don't worry about reviewing that for now.

Good callouts on the styling! I'll see what I can do to make it look closer to the current button.

**Comment by campionfellin** [comment]
New screenshot:


Screen Shot 2020-12-03 at 6 11 41 PM

**Comment by campionfellin** [comment]
Hey @frboulais any other big comments on this one?

File: Bundle.swift [hunk 1]
```diff
@@ -10,4 +10,6 @@ import Foundation
 
 internal extension Bundle {
     static let module = Bundle(for: BMCManager.self)
+    @available(iOS 13.0, *)
+    static let moduleUI = Bundle(for: BMCManagerUI.self)
```
Comment (frboulais) [inline Bundle.swift]: Why adding another module, it should be the same, no?

File: BuyMeACoffee.podspec [hunk 2]
```diff
@@ -20,7 +20,7 @@ Pod::Spec.new do |spec|
   spec.source_files          = 'Sources/**/*.swift', 'Bundle.swift'
   spec.resources             = 'Sources/**/Resources/*'
 
-  spec.ios.framework         = 'UIKit', 'WebKit'
+  spec.ios.framework         = 'UIKit', 'WebKit', 'SwiftUI'
```
Comment (frboulais) [inline BuyMeACoffee.podspec:23]: 👍

File: Sources/BuyMeACoffee/BMCColor.swift [hunk 3]
```diff
@@ -21,27 +36,46 @@ public enum BMCColor: String, CaseIterable {
     
     public static let `default`: Self = .yellow
     
-    internal var background: UIColor? {
-        UIColor(named: rawValue, in: .module, compatibleWith: nil)
+    internal var background: CustomColor? {
+        if #available(iOS 13.0, OSX 10.15, *) {
+            return Color(UIColor(named: rawValue, in: .module, compatibleWith: nil)!)
```
Comment (frboulais) [inline Sources/BuyMeACoffee/BMCColor.swift]: Why does it return either a `Color` or `UIColor` depending the platform?

File: Sources/BuyMeACoffee/BMCColor.swift [hunk 4]
```diff
@@ -7,6 +7,21 @@
 
 #if canImport(UIKit)
 import UIKit
+#if canImport(SwiftUI)
+import SwiftUI
+#endif
+
+protocol CustomColor { }
+
+extension UIColor: CustomColor { }
```
Comment (frboulais) [inline Sources/BuyMeACoffee/BMCColor.swift]: These empty protocols are really not a good way to handle the SwiftUI/UIKit thing

File: Sources/BuyMeACoffee/BMCButtonUI.swift [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    
+    public var body: some View {
+        VStack {
+            Button(action: {
+                BMCManagerUI.shared.start()
+                let payment = SKPayment(product: BMCManagerUI.shared.product!)
+                SKPaymentQueue.default().add(payment)
+            }) {
+                self.image.padding(EdgeInsets(top: 0, leading: -6, bottom: 0, trailing: 6))
+                Text(self.title!).padding(EdgeInsets(top: 0, leading: 6, bottom: 0, trailing: -6)).font(configuration.font.value as? Font).foregroundColor(configuration.color.title as? Color)
```
Comment (frboulais) [inline Sources/BuyMeACoffee/BMCButtonUI.swift]: Unreadable indentation

**Review by frboulais — CHANGES_REQUESTED** [review]
Thank you for you contribution, but I can't approve these changes.
Both the code & the button don't fit the framework & Buy Me a Coffee guidelines.
Too much duplicated code, too much `#if platform` in the code. 
You can still use it as it is from your fork. I'll find someone to help you with that SwiftUI support :)

**Comment by campionfellin** [comment]
Thanks for the feedback! I know there's a lot of issues with it, but this really helps! I'll continue iterating

File: Sources/BuyMeACoffee/BMCColor.swift (hunk 3 above)
Comment (campionfellin) [reply Sources/BuyMeACoffee/BMCColor.swift]: Good callout, no need to! Fixed.

**Comment by campionfellin** [comment]
Getting fairly close here. Pared most changes down to just the new `BMCButtonUI` file

**Pull request closed without merging**